src/llvm/lib_llvm.cr

```diff
@@ -1,20 +1,36 @@
 {% begin %}
-lib LibLLVM
-  LLVM_CONFIG = {{ env("LLVM_CONFIG") || `#{__DIR__}/ext/find-llvm-config`.stringify }}
-end
-{% end %}
+  {% if flag?(:win32) && flag?(:preview_dll) %}
+    {% config = nil %}
+    {% for dir in Crystal::LIBRARY_PATH.split(';') %}
+      {% config ||= read_file?("#{dir.id}/llvm_VERSION") %}
+    {% end %}
 
-{% begin %}
-  {% unless flag?(:win32) %}
-    @[Link("stdc++")]
+    {% unless config %}
+      {% raise "Cannot determine LLVM configuration; ensure the file `llvm_VERSION` exists under `CRYSTAL_LIBRARY_PATH`" %}
+    {% end %}
+
+    {% lines = config.lines.map(&.chomp) %}
+    {% llvm_version = lines[0] %}
+    {% llvm_targets = lines[1] %}
+    {% llvm_ldflags = lines[2] %}
+    {% extra_lib = "llvm" %}
```

Comment:
**nitpick:** The `extra_lib` variable seems unnecessary. We could just write the link annotations directly in its stead.
```suggestion
    @[Link("llvm")]
```